classes/notification/managerDelegate/ProductionStatusNotificationManager.inc.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * @file classes/notification/managerDelegate/ProductionStatusNotificationManager.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ProductionStatusNotificationManager
+ * @ingroup classses_notification_managerDelegate
+ *
+ * @brief Production status notifications types manager delegate.
+ */
+
+import('lib.pkp.classes.notification.NotificationManagerDelegate');
+
+class ProductionStatusNotificationManager extends NotificationManagerDelegate {
+
+	/**
+	 * Constructor.
+	 * @param $notificationType int NOTIFICATION_TYPE_...
+	 */
+	function ProductionStatusNotificationManager($notificationType) {
+		parent::NotificationManagerDelegate($notificationType);
+	}
+
+	/**
+	 * @copydoc PKPNotificationOperationManager::getNotificationMessage()
+	 */
+	public function getNotificationMessage($request, $notification) {
+		switch ($notification->getType()) {
+			case NOTIFICATION_TYPE_ASSIGN_PRODUCTIONUSER:
+				return __('notification.type.assignProductionUser');
+			case NOTIFICATION_TYPE_AWAITING_REPRESENTATIONS:
+				return __('notification.type.awaitingRepresentations');
+		}
+	}
+
+	/**
+	 * @copydoc PKPNotificationOperationManager::getNotificationUrl()
+	 */
+	public function getNotificationUrl($request, $notification) {
+		$url = parent::getNotificationUrl($request, $notification);
+		$dispatcher = Application::getDispatcher();
+		$contextDao = Application::getContextDAO();
+		$context = $contextDao->getById($notification->getContextId());
+
+		switch ($notification->getType()) {
+			case NOTIFICATION_TYPE_ASSIGN_PRODUCTIONUSER:
+			case NOTIFICATION_TYPE_AWAITING_REPRESENTATIONS:
+				assert($notification->getAssocType() == ASSOC_TYPE_SUBMISSION && is_numeric($notification->getAssocId()));
+				return $dispatcher->url($request, ROUTE_PAGE, $context->getPath(), 'workflow', 'access', $notification->getAssocId());
+		}
+	}
+
+	/**
+	 * @copydoc PKPNotificationOperationManager::getStyleClass()
+	 */
+	public function getStyleClass($notification) {
+		return NOTIFICATION_STYLE_CLASS_INFORMATION;
+	}
+
+	/**
+	 * @copydoc NotificationManagerDelegate::updateNotification()
+	 */
+	public function updateNotification($request, $userIds, $assocType, $assocId) {
+		$context = $request->getContext();
+		$contextId = $context->getId();
+
+		assert($assocType == ASSOC_TYPE_SUBMISSION);
+		$submissionId = $assocId;
+		$submissionDao = Application::getSubmissionDAO();
+		$submission = $submissionDao->getById($submissionId);
+
+		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+		$editorStageAssignments = $stageAssignmentDao->getEditorsAssignedToStage($submissionId, $submission->getStageId());
+
+		// One way to check if a production user is assigned
+		// Get the 'layout assignment' notifications
+		$layoutEditorAssigned = false;
```

Comment:
As above: suggest making this only dependent on queries, not notifications. And since that will make these two blocks of code basically identical, maybe they could be rolled into one?